resolve-ref/action.yml

```diff
@@ -0,0 +1,85 @@
+# Copyright 2024 Kinetic Commerce
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Resolve a ref-ish value
+description: |
+  Resolves a provided open pull request, branch, tag, or commit ref to
+  a valid commit reference.
+
+  Ref resolution is always in the order of `pr`, `branch`, `tag`, `commit`,
+  `default-branch`, `repo-default-branch`.
+author: 'KineticCafe'
+inputs:
+  token:
+    description: |
+      The GitHub token for this action.
+    required: true
+    default: ${{ github.token }}
+  ref:
+    description: |
+      The branch name, open pull request number, tag, or commit reference to
+      resolve.
+
+      Pull request numbers must always begin with `#` to distinguish them from
+      commits that are all decimal digits. Closed or merged pull requests will
+      be ignored.
+
+      If missing or empty, this action will fall back to the resolved value of
+      `default-branch`.
+    required: false
+  default-branch:
+    description: |
+      The optional branch name to for resolving the default ref.
+
+      If not provided or empty, falls back to the repository default branch.
+    required: false
+  allowed:
+    description: |
+      The list of ref types supported in this call. Typically, this will be
+      used to restrict `ref` to open pull requests (with `pr`). This is
+      a comma-delimited string that defaults to `pr,branch,tag,commit`.
```

Comment:
(nit) comma-separated or comma-delimted?

This is a string containing a comma separated list of ref types to be considered in this call.

Reply:
> This is a comma-delimited list of `ref` types that are permitted for resolution.